include/fw_info.rst

```diff
@@ -0,0 +1,123 @@
+It includes the following information:
+
+* The size of the firmware image.
+* The single, monotonically increasing version number of the image.
+* The address through which to boot into the firmware (the vector table address).
+  This address is not necessarily the start of the image.
+
+Additionally, there is information for exchanging arbitrary data:
+
+* Application Binary Interface (ABI) getter (:cpp:member:`abi_out`)
+* Pointer to ABI getter (:cpp:member:`abi_in`)
+
+ABIs
+****
+
+The firmware information structure allows for exchange of arbitrary tagged and versioned interfaces called Application Binary Interfaces (ABIs).
+
+An ABI structure is a structure consisting of a header followed by arbitrary data.
+The header consists of the following information:
+
+* ABI ID (uniquely identifies the ABI)
+* ABI version (single, monotonically increasing version number for the ABI with this ID)
+* ABI flags (32 individual bits for indicating the particulars of the ABI)
+* ABI length (length of the following data)
+
+To retrieve an ABI, a firmware image calls another firmware image's ABI getter.
+Every image must provide an ABI getter (:cpp:member:`abi_out`) that other images can use to retrieve its ABIs.
+This ABI getter is a function pointer that retrieves ABI structs (or rather pointers to ABI structs).
+
+In addition, every image can access other ABIs through a second ABI getter (:cpp:member:`abi_in`).
+This ABI getter must be provided when booting the image.
+In other words, an image should expect its :cpp:member:`abi_in` ABI getter to be filled at the time it boots, and will not touch it during booting.
+After booting, an image can call :cpp:member:`abi_in` to retrieve ABIs from the image or images that booted it without knowing where they are located.
+
+Each image can provide multiple ABIs.
+An ABI getter function takes an index, and each index from 0 to *n* must return a different ABI, given that the image provides *n* ABIs with the same ID.
+
+Typically, the actual ABI will be a function pointer (or a list of function pointers), but the data can be anything, though it should be considered read-only.
+The reason for making the ABI getters function pointers instead of pointing to a list of ABIs is to allow dynamic creation of ABI lists without using RAM.
+
+Usage
+*****
+
+To locate and verify firmware info structures, use :cpp:func:`fw_info_find` and :cpp:func:`fw_info_check`, respectively.
+
+To find an ABI with a given version and flags, call :cpp:func:`fw_info_abi_find`.
+This function calls :cpp:member:`abi_in` under the hood, checks the ABI's version against the allowed range, and checks that it has all the flags set.
+
+To populate an image's :cpp:member:`abi_in` (before booting the image), the booting image should call :cpp:func:`fw_info_abi_provide` with its own firmware information structure.
```

Comment:
"with its own firmware information structure" = the booting image's
Is that correct?

Reply:
No, it's the other image's information structure :). I corrected it in the "doc fixup" commit.